### PR TITLE
score

### PR DESCRIPTION
**Pull request by matthewdu**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,11 +66,31 @@ function tileSolveRequest(reqObj, socket){
...
+		io.emit(phase, score);
 		setTimeout(setupPhase, 5000);
 	} else {
 		console.log('There are still more moves: ' + user);
 		socket.emit('errorNoMoreMovesRequest');
+		if(score[reqUser.username]){
+			score[reqUser.username].points -= 1;
+			if(score[reqUser.username].points < 0){
+				score[reqUser.username].points = 0;
+			}
```
Comment (dvdfu) [inline index.js]: why not do `points -= 1` after a `points > 0` check and then use `else`? Saves one write operation

File: index.js [hunk 2]
```diff
@@ -18,12 +19,12 @@ function User(username, thumbnail){
...
-        return users[i];
-    }
-  }
-  return -1;
+	for(var i = 0; i < users.length; i++){
+		if(users[i].username == name){
+			return users[i];
+		}
+	}
+	return -1;
```
Comment (Nosajool) [inline index.js:27]: return the user or -1?

File: index.js [hunk 3]
```diff
@@ -42,6 +43,14 @@ function tileSolveRequest(reqObj, socket){
...
 			console.log('Answer found by user: ' + JSON.stringify(reqUser))
 			correctReq = true;
+			if(score[reqUser.username]){
+				score[reqUser.username].points += 1;
+			} else {
+				score[reqUser.username] = {
+					user: reqUser,
+					points: 1
+				};
+			}
```
Comment (Nosajool) [inline index.js:53]: test this. Does it work in javascript?

File: index.js [hunk 4]
```diff
@@ -59,11 +68,30 @@ function tileSolveRequest(reqObj, socket){
...
 		console.log('No more moves found by user: ' + user);
-		io.emit('setupPhase');
+		if(score[reqUser.username]){
+			score[reqUser.username].points += 2;
+		} else {
+			score[reqUser.username] = {
+				user: reqUser,
+				points: 2
+			};
+		}
```
Comment (Nosajool) [inline index.js]: same as above

File: index.js [hunk 5]
```diff
@@ -72,6 +100,7 @@ function setupPhase() {
 	answers = tileGenerator.solveTiles(tiles);
 	console.log(answers)
 	moveLog = [];
+	score = {};
```
Comment (Nosajool) [inline index.js:107]: resetting the score every round?

File: index.js [hunk 6]
```diff
@@ -84,8 +113,6 @@ app.get('/', function(req, res){
 app.use(express.static(__dirname+'/public'));
 
 io.on('connection', function(socket){
-	socket.emit('userSetup');
```
Comment (Nosajool) [inline index.js]: !!!!!!!!!!!

**Pull request merged**